interface/network_interface.cpp

```diff
@@ -0,0 +1,169 @@
+#include <cstring>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <net/if.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+
+#include <sys/ioctl.h>
+
+#include "network_interface.h"
+#include "ilogger.h"
+#include "interface_addresses.h"
+
+NetworkInterface::NetworkInterface(const std::string& name, std::shared_ptr<ILogger> logger) :
+        m_logger(logger),
+        m_name(name)
+{
+    m_fd = socket(AF_INET, SOCK_DGRAM, 0);
+}
+
+NetworkInterface::~NetworkInterface()
+{
+    if (m_fd >= 0)
+    {
+        close(m_fd);
+    }
+}
+
+bool NetworkInterface::AddIpAddress(const std::string& ipAddress, const std::string& netmask) const
+{
+    if (m_fd < 0)
+    {
+        perror("socket");
+        LOG_ERROR(m_logger, "socket fd is negative: " + std::to_string(m_fd));
+        return false;
+    }
+
+    uint32_t ip_address = network_uint32_ipv4_from_string(ipAddress);
+    if (ip_address == 0)
+    {
+        return false;
+    }
+    uint32_t netmask_bin = network_uint32_ipv4_from_string(netmask);
+    if (netmask_bin == 0)
+    {
+        return false;
+    }
+
+    if (!set_ip_address(ip_address))
+    {
+        return false;
+    }
+    LOG_INFO(m_logger, "successfully set ip: " + ipAddress + " on interface: " + m_name);
```

Comment:
Personal taste kind of thing : relevant for many logs - I thinks it's better to start with capital letter for all logs. i.e.: 
```suggestion
    LOG_INFO(m_logger, "Successfully set ip: " + ipAddress + " on interface: " + m_name);
```